File: src/008/main.py
from functools import reduce
from sys import argv

N_DIGITS = 13
largest = 0
current = [0] * N_DIGITS

with open (argv [1]) as file:
	NUMBER = file.read () [:-1]

for char in NUMBER:
	current = current [1:] + [int (char)]
	product = reduce (lambda a, b: a * b, current)
	
	if product > largest:
		largest = product

print (largest)

File: src/008/main.rb

N_DIGITS = 13
NUMBER = File.read ARGV.first

largest = 0

NUMBER.chars.reduce [0] * N_DIGITS do |current, i|
	current = current [1..] + [i.to_i]
	product = current.reduce :*
	largest = product if product > largest
	current
end

puts largest

